### ISSUE TITLE
Excluded Unimodule will persistently be included

### ISSUE THREAD

**Issue by kennetpostigo** [expo/expo#11742]
### Summary of Issue 
I'm trying to exclude the `expo-location` module from being added as my app keeps getting rejected from the playstore because of location permissions. However I've removed the auto included location permissions that were added when scaffolding a react-native unimodules app.

## Unimodules configuration

```json
  "react-native-unimodules": {
    "ios": {
      "exclude": [
        "expo-location"
      ]
    },
    "android": {
      "exclude": [
        "expo-location"
      ]
    }
  },
```

This line keeps getting added to `BasePackageList`:
```java
new expo.modules.location.LocationPackage(),
```

I think this module seems to define location permissions, i'm not quite sure what else is requiring the location permissions.

Heres the portion of the AndroidManifest file:

```xml
  <!-- OPTIONAL PERMISSIONS, REMOVE WHATEVER YOU DO NOT NEED -->
  <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
  <uses-permission android:name="android.permission.READ_PHONE_STATE"/>
  <uses-permission android:name="android.permission.USE_FINGERPRINT"/>
  <uses-permission android:name="android.permission.USE_BIOMETRIC"/>
  <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS"/>
  <!-- These require runtime permissions on M -->
  <!-- <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
  <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
  <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION"/> -->
```

**Comment by brentvatne** [comment]
https://docs.expo.io/versions/latest/sdk/permissions/#permissions-on-android

**Comment by Bardiamist** [comment]
@brentvatne I noticed that "exclude" don't work in gradle. It works with "package.json". It was hard to find. So documentation is wrong now https://docs.expo.io/bare/installing-unimodules/#-need-to-exclude-some-unimodules-that

**Comment by brentvatne** [comment]
hmm it should be supported: https://github.com/expo/expo/blob/master/packages/react-native-unimodules/gradle.groovy#L189

can you create a [mcve](https://stackoverflow.com/help/minimal-reproducible-example) that demonstrates this?

**Comment by Bardiamist** [comment]
@brentvatne ok, repro here: https://github.com/Bardiamist/unimod

`addUnimodulesDependencies([exclude: ['expo-constants']])` removing `expo-constants` from `getPackageList` but `expo-constants` still in the project:

<img width="233" alt="Screenshot 2021-05-04 at 11 57 11" src="https://user-images.githubusercontent.com/22082342/116962340-e67b9a80-accf-11eb-80a3-9fed124274e1.png">

`expo-constants` will disappear if add
```
"react-native-unimodules": {
    "android": {
      "exclude": [
        "expo-constants"
      ]
    }
  }
  ```
  and sync gradle.

**Comment by brentvatne** [comment]
@tsapeta - can you investigate?

**Comment by tsapeta** [comment]
Hey @Bardiamist,

If you want to pass autolinking options in Gradle, you should do this twice — in both `addUnimodulesDependencies` (build.gradle) and `includeUnimodulesProjects` (settings.gradle). Because of the way how these two files are being executed and resolved by Gradle, we can't share the options between these methods other than specifying the options in `package.json`.

Also, don't be afraid that something is still visible in Android Studio — it shows all locally included projects (based on settings.gradle) but the project will not be bundled into your app as long as nothing else depends on it (e.g. when it's excluded in `addUnimodulesDependencies`).

To sum up, you have three options:
- Leave it as is and let the projects be **included** (settings.gradle), but not **added** (build.gradle)
- Use the same options across these two files
- Keep one configuration in package.json (I recommend it)

**Comment by Bardiamist** [comment]
> and `includeUnimodulesProjects` 

Need to add it to readme then. Or better about `package.json`.

> Also, don't be afraid that something is still visible in Android Studio — it shows all locally included projects (based on settings.gradle) but the project will not be bundled into your app as long as nothing else depends on it (e.g. when it's excluded in addUnimodulesDependencies).

My build was failed with `expo-constants` until I excluded it in `package.json`
